ksql-engine/src/test/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNodeTest.java

```diff
@@ -164,78 +201,159 @@ public void shouldHaveCorrectOutputNodeSchema() {
     assertThat(keyField, equalTo(new Field("field2", 1, Schema.OPTIONAL_STRING_SCHEMA)));
     assertThat(schemaKStream.getSchema().fields(), equalTo(schema.fields()));
   }
 
   @Test
   public void shouldCreateSinkTopic() {
-    EasyMock.verify(topicClient);
-  }
-
-  private SchemaKStream buildStream() {
-    builder = new StreamsBuilder();
+    // When:
+    stream = buildStream();
 
-    return outputNode.buildStream(builder,
-        ksqlConfig,
-        serviceContext,
-        new InternalFunctionRegistry(),
-        new HashMap<>());
+    // Then:
+    verify(topicClient).createTopic(SINK_KAFKA_TOPIC_NAME, 4, (short) 3, Collections.emptyMap());
   }
 
+
   @Test
   public void shouldCreateSinkWithCorrectCleanupPolicyNonWindowedTable() {
+    // Given:
     outputNode = getKsqlStructuredDataOutputNode(Serdes.String());
 
-    final KafkaTopicClient topicClientForNonWindowTable = EasyMock.mock(KafkaTopicClient.class);
-    final Map<String, String> topicConfig = ImmutableMap.of(
-        TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT);
-    topicClientForNonWindowTable.createTopic("output", 4, (short) 3, topicConfig);
-    EasyMock.replay(topicClientForNonWindowTable);
-
+    // When:
     final SchemaKStream schemaKStream = buildStream();
 
+    // Then:
+    final Map<String, String> topicConfig = ImmutableMap.of(
+        TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT);
+    verify(topicClient).createTopic(SINK_KAFKA_TOPIC_NAME, 4, (short) 3, topicConfig);
     assertThat(schemaKStream, instanceOf(SchemaKTable.class));
   }
 
   @Test
   public void shouldCreateSinkWithCorrectCleanupPolicyWindowedTable() {
+    // Given:
     outputNode = getKsqlStructuredDataOutputNode(
         WindowedSerdes.timeWindowedSerdeFrom(String.class));
 
-    final KafkaTopicClient topicClientForWindowTable = EasyMock.mock(KafkaTopicClient.class);
-    topicClientForWindowTable.createTopic("output", 4, (short) 3, Collections.emptyMap());
-    EasyMock.replay(topicClientForWindowTable);
-
+    // When:
     final SchemaKStream schemaKStream = buildStream();
 
+    // Then:
+    verify(topicClient).createTopic(SINK_KAFKA_TOPIC_NAME, 4, (short) 3, Collections.emptyMap());
     assertThat(schemaKStream, instanceOf(SchemaKTable.class));
   }
 
   @Test
   public void shouldCreateSinkWithCorrectCleanupPolicyStream() {
-    final KafkaTopicClient topicClientForWindowTable = EasyMock.mock(KafkaTopicClient.class);
-    topicClientForWindowTable.createTopic("output", 4, (short) 3, Collections.emptyMap());
-    EasyMock.replay(topicClientForWindowTable);
-
+    // When:
     final SchemaKStream schemaKStream = buildStream();
 
+    // Then:
+    verify(topicClient).createTopic(SINK_KAFKA_TOPIC_NAME, 4, (short) 3, Collections.emptyMap());
+    assertThat(schemaKStream, instanceOf(SchemaKStream.class));
+  }
+
+  @Test
+  public void shouldCreateSinkWithTheSourcePartititionReplication() {
```

Comment:
Ensure we have tests that cover:
- Have just partition count in with clause, and that sink is created with correct values.
- Have just replica count in with clause, and that sink is created with correct values.
- Have just both in with clause, and that sink is created with correct values.
- Have just neither in with clause, and that sink is created with correct values.
- Have one/both in with clause, and sink has different values - what should happen? Throw? Ignore?
- Have neither in with clause, and sink has different values to source - what should happen? Throw? Ignore?

Reply:
There was one test that asserted fail when the properties are not correct. I splited it into two tests where it explicitly tests partition count and replica count.

Reply:
Hm, the version I'm looking at has only one test (`shouldThrowIfSinkTopicHasDifferentPropertiesThanRequested`) but it looks good to me! Thanks for following up on this.